### ISSUE TITLE
CVODE_BDF: Statistics is wrong if zero crossings occur

### ISSUE THREAD

**Issue by MartinOtter** [SciML/DifferentialEquations.jl#763]
If CVODE_BDF is used with zero crossing functions and state events occur, then the reported statistics is wrong ( sol.destats.naccept,  sol.destats.nreject, sol.destats.nf, sol.destats.njacs,  sol.destats.nreject ), for example njacs=1, although 20 state events occurred. Most likely, the statistics data is not accumulated, but reset to zero when the integration is restarted after an event occurred.

**Comment by ChrisRackauckas** [comment]
Yeah we've noticed that Sundials does something funny with its stats, but never really figured out how to correct it.

**Comment by ChrisRackauckas** [comment]
Maybe whenever we `reinit!` we need to add the old stats onto it.
